lib/wallaby/experimental/selenium.ex

```diff
@@ -1,40 +1,102 @@
+
+  ```
+  {:ok, session} = Wallaby.start_session()
+  ```
+
+  ## Configuration
+
+  ### Capabilities
+
+  These capabilities will override the default capabilities.
+
+  ```
+  config :wallaby,
+    selenium: [
+      capabilities: %{
+        # something
+      }
+    ]
+  ```
+
+  ## Default Capabilities
+
+  By default, Selenium will use the following capabilities
+
+  You can read more about capabilities in the [JSON Wire Protocol](https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#capabilities-json-object) documentation.
+
+  ```elixir
+  %{
+    javascriptEnabled: true,
+    browserName: "firefox",
+    "moz:firefoxOptions": %{
+      args: ["-headless"]
+    }
+  }
+  ```
+
+  ## Notes
+
+  - Requires [selenium-server-standalone](https://www.seleniumhq.org/download/) to be running on port 4444. Wallaby does _not_ manage the start/stop of the Selenium server.
+  - Requires [GeckoDriver](https://github.com/mozilla/geckodriver) to be installed in your path when using [Firefox](https://www.mozilla.org/en-US/firefox/new/). Firefox is used by default.
+  """
+
   use Supervisor
 
   @behaviour Wallaby.Driver
 
   alias Wallaby.{Driver, Element, Session}
   alias Wallaby.Experimental.Selenium.WebdriverClient
 
+  @typedoc """
+  Options to pass to Wallaby.start_session/1
+
+  ```elixir
+  Wallaby.start_session(
+    remote_url: "http://selenium_url",
+    capabilities: %{browserName: "firefox"},
+    create_session_fn: fn url, capabilities ->
+      WebdriverClient.create_session(url, capabilities)
+    end
+  )
+  ```
+  """
   @type start_session_opts ::
-    {:remote_url, String.t} |
-    {:capabilities, map} |
-    {:create_session_fn, ((String.t, map) -> {:ok, %{}})}
+          {:remote_url, String.t()}
+          | {:capabilities, map}
+          | {:create_session_fn, (String.t(), map -> {:ok, %{}})}
 
+  @typedoc false
+  @type end_session_opts :: {:end_session_fn, (Session.t() -> any)}
+
+  @doc false
   def start_link(opts \\ []) do
     Supervisor.start_link(__MODULE__, :ok, opts)
   end
 
+  @doc false
   def init(:ok) do
-    children = [
-    ]
-
-    supervise(children, strategy: :one_for_one)
+    supervise([], strategy: :one_for_one)
   end
 
+  @doc false
   def validate do
     :ok
   end
 
-  @spec start_session([start_session_opts]) :: {:ok, Session.t}
+  @doc false
+  @spec start_session([start_session_opts]) :: Wallaby.Driver.on_start_session() | no_return
   def start_session(opts \\ []) do
     base_url = Keyword.get(opts, :remote_url, "http://localhost:4444/wd/hub/")
-    capabilities = Keyword.get(opts, :capabilities, %{})
-    create_session_fn = Keyword.get(opts, :create_session_fn,
-                                    &WebdriverClient.create_session/2)
+    create_session_fn = Keyword.get(opts, :create_session_fn, &WebdriverClient.create_session/2)
 
-    capabilities = Map.merge(default_capabilities(), capabilities)
+    capabilities = Keyword.get(opts, :capabilities, capabilities_from_config())
```

Comment:
Awesome, thank you! So first we try to get capabilities from opts, then from application config, then take the default ones. This is great!